### PR TITLE
fix(data-table): pass column key to onRowClick callback

### PR DESCRIPTION
**Pull request by jonnybel**
#### Summary

This PR fixes the DataTable `onRowClick` callback to pass the column key as an argument of the method.

On the "old" table, this information was passed to the method, but on the new one, it was removed as an `onClick` method was available per-column. However, the `onClick` method was later [also removed](https://github.com/commercetools/ui-kit/pull/1314) with no logical replacement, which made it hard to define per-column specific behaviours for `onClick` events on the Cells.

### CODE REVIEW

File: packages/components/data-table/src/data-row.js [hunk 1]
```diff
@@ -53,6 +47,13 @@ const DataRow = (props) => {
             props.columns.length,
             columnIndex
           )}
+          onCellClick={
```
Comment (jonnybel) [inline packages/components/data-table/src/data-row.js:50]: The `onClick` event handler had to be moved from the Row to each of the DataCells here in order to have the `column` information available.

File: packages/components/data-table/src/data-row.js [hunk 2]
```diff
@@ -53,6 +47,13 @@ const DataRow = (props) => {
...
+            props.onRowClick && !column.shouldIgnoreRowClick
+              ? () =>
+                  props.onRowClick &&
+                  props.onRowClick(props.row, props.rowIndex, column.key)
```
Comment (tdeekens) [inline packages/components/data-table/src/data-row.js:54]:
```suggestion
                  props.onRowClick?.(props.row, props.rowIndex, column.key)
```

Should work with optional chaining 🤔 . Syntax looks really odd but I think it was it.

Comment (jonnybel) [reply packages/components/data-table/src/data-row.js]: Ah nice! I'm committing this and let the tests be the judge :)

**Comment by jonnybel** [comment]
@adnasa I believe this would help to solve a regression issue that you mentioned in the MC.

**Comment by adnasa** [comment]
please ping me when this is in and I'll update MC-fe

**Pull request merged**